### ISSUE TITLE
bower2nix generates the wrong hash

### ISSUE THREAD

**Issue by pikajude** [NixOS/nixpkgs#8912]
With the following `bower.json`:

``` json
{
  "name": "test",
  "version": "0.0.0",
  "dependencies": {
    "tipsy": "*"
  }
}
```

```
$ nix-build -E "with import <nixpkgs> {}; pkgs.callPackage ($(bower2nix bower.json /dev/stdout)) {}"
these derivations will be built:
  /nix/store/f494jasghb1nhdcgr0yi4nwpdi10lxxi-tipsy-0.1.7.drv
  /nix/store/3m0i74dl3p64dmnyg06vs51h9h3c8dd9-bower-env.drv
building path(s) ‘/nix/store/4r5sjdv3vk0nlhv7iywf5frnb894v1vv-tipsy-0.1.7’
{"name":"tipsy","homepage":"https://github.com/jaz303/tipsy","version":"0.1.7","_release":"0.1.7","_resolution":{"type":"version","tag":"v0.1.7","commit":"57ec6ab3479e5af820fdacf61770a8f5f5c33878"},"_source":"git://github.com/jaz303/tipsy.git","_target":"0.1.7"}output path ‘/nix/store/4r5sjdv3vk0nlhv7iywf5frnb894v1vv-tipsy-0.1.7’ should have r:sha256 hash ‘01kvfmmq75s6dwpr0f1i0i41lpj2vf9sqf4c4xmm6kaw53jdxiij’, instead has ‘0ldkxrl2vklw40afji09kmq1qqw71w1df77fp0bsv8ndnhnrdhr8’
cannot build derivation ‘/nix/store/3m0i74dl3p64dmnyg06vs51h9h3c8dd9-bower-env.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/3m0i74dl3p64dmnyg06vs51h9h3c8dd9-bower-env.drv’ failed
```

**Comment by pikajude** [comment]
@shlevy is it possible this is `bower2nix`'s fault and not a problem in nixpkgs?

**Comment by shlevy** [comment]
Definitely possible, I haven't touched bower2nix in a long time and it was just a "minimum effort to get the work done" kind of project.

**Comment by rvl** [comment]
Hi @pikajude. I am preparing a PR for an updated `bower2nix` in nixpkgs. In the mean time, a new version can be installed from here: https://github.com/rvl/bower2nix. It seems to work for that `bower.json`.

**Comment by rvl** [comment]
After further investigation it seems to be because of some symlinks in the `tipsy` package. I think it was fixed by commit b7155460719ba2c8b0b276fad3fdd22441eac9f9 (@rbvermaa).

**Comment by abuibrahim** [comment]
this might be a duplicate of #18454 fixed by PR #18569

**Comment by rvl** [comment]
Thanks @abuibrahim I'm checking your PR now.

I think this issue can be closed because there's a new version of bower2nix in nixpkgs which probably fixes it. If not, an issue can be raised at https://github.com/rvl/bower2nix/ .
